flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/generated/CompileUtils.java

```diff
@@ -59,14 +62,16 @@
 	 * @return  the compiled class
 	 */
 	public static <T> Class<T> compile(ClassLoader cl, String name, String code) {
-		Tuple2<ClassLoader, String> cacheKey = Tuple2.of(cl, name);
-		Class<?> clazz = COMPILED_CACHE.getIfPresent(cacheKey);
-		if (clazz == null) {
-			clazz = doCompile(cl, name, code);
-			COMPILED_CACHE.put(cacheKey, clazz);
+		try {
+			Cache<ClassLoader, Class> compiledClasses = COMPILED_CACHE.get(name,
+					() -> CacheBuilder.newBuilder().maximumSize(5).weakKeys().softValues().build());
+			return compiledClasses.get(cl, () -> doCompile(cl, name, code));
+		} catch (ExecutionException | UncheckedExecutionException e) {
```

Comment:
Why do we need to catch these special exceptions?

Reply:
We're supposed to throw `InvalidProgramException` if the compilation fails (`CompileUtilsTest::testWrongCode` verifies that). So we have to catch the execution exceptions and check the cause.

Reply:
I don't think we have to do this just for testing. 
We can just `throw new FlinkRuntimeException(e.getMessage(), e);` in this clause and expect `FlinkRuntimeException` exception in the test.

Reply:
OK, although I think the test shouldn't be expecting any specific kind of exception because exception is not a part of the method contract.

Reply:
NIT: I would prefer removing guava wrapped exceptions. It's widely used.

Reply:
@JingsongLi We're catching `Exception` now

Reply:
I mean remove guava wrapped exceptions in the stack as you previous did.

Reply:
What's the benefit? From my point of view, it will introduce some complex codes.

Reply:
It is just nit suggestion.
Because `ExecutionException` is an invalid exception, and it could/should be threw. This can simplify exception stack.
You can take a look to: https://github.com/apache/flink/blob/4adf23d3b93c301b53f11f1c05a1aa6d37112d46/flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/taskmanager/AbstractTaskManagerFileHandler.java#L125
It is an example. And there are some examples in Spark too.

Reply:
Thanks for the link. I'm to strip `ExecutionException` to simplify exception stack.